lib/puppet/parser/resource.rb

```diff
@@ -184,6 +184,15 @@ def name
   # if we ever receive a parameter named 'tag', set
   # the resource tags with its value.
   def set_parameter(param, value = nil)
+    if evaluated?
+      strict = Puppet[:strict]
+      if :off != strict
+        msg = _('Attempt to override an already evaluated resource %{resource}') % { resource: ref }
```

Comment:
A problem here is that there is no reference to any file or line number. How is the user supposed to find where the problem is? The error needs to come from where the override is executing as it should have a reference to the override expression causing this problem. Here, only the mutated resource is know which is of some, but in comparison much lower value.